File: csnet/commons/src/org/vpc/neormf/commons/sql/SqlParam.cs
using System;
using System.Data;
using System.Data.SqlClient;
using org.vpc.neormf.commons.jwrapper;
using org.vpc.neormf.commons.util;
/**
 * class presentation
 *
 * @author taha BEN SALAH (tbsalah)
 * @version 1.0
 * @copyrights (c) 2004, Vpc Open Source Foundary
 * @project Root Optimizer
 * @creation on Date: 28 juil. 2004 Time: 11:20:37
 * @modification on ---- by -----
 * @modification on ---- by -----
 * @modification on ---- by -----
 */
namespace org.vpc.neormf.commons.sql
{
	public class SqlParam : SqlStatementParamsProvider 
						  {
    private int pos;
    private String name;
    private Object value;
    private int sqlType;

    public SqlParam() {
    }

    public int GetPos() {
        return pos;
    }

    public void SetPos(int pos) {
        this.pos = pos;
    }

    public String GetName() {
        return name;
    }

    public void SetName(String name) {
        this.name = name;
    }

    public Object GetValue() {
        return value;
    }

    public void SetValue(Object value) {
        this.value = value;
    }

    public int GetSqlType() {
        return sqlType;
    }

    public void SetSqlType(int sqlType) {
        this.sqlType = sqlType;
    }

    public int PopulateStatement(PreparedStatement ps,int startPosition){
        int rpos=pos+startPosition-1;
        if (value == null) {
            ps.setNull(rpos, sqlType);
            return 1;
        }
        switch (sqlType) {
            case Types.VARCHAR:
                {
                    ps.SetString(rpos, (String) value);
                    break;
                }
            case Types.INTEGER:
                {
                    ps.SetInt(rpos, ((int) value));
                    break;
                }
            case Types.DOUBLE:
                {
                    ps.SetDouble(rpos, ((double) value));
                    break;
                }
            case Types.SMALLINT:
                {
                    ps.setShort(rpos, ((short) value));
                    break;
                }
            case Types.TIME:
                {
                    ps.SetTime(rpos, ((TimeSpan) value));
                    break;
                }
            case Types.TIMESTAMP:
                {
                    ps.SetTimestamp(rpos, ((DateTime) value));
                    break;
                }
            case Types.BIGINT:
                {
                    ps.SetLong(rpos, ((long) value));
                    break;
                }
            case Types.FLOAT:
                {
                    ps.SetFloat(rpos, ((float) value));
                    break;
                }
            case Types.DATE:
                {
                    ps.SetDate(rpos, ((DateTime) value));
                    break;
                }
            default:
                {
                    throw new Exception("Unhandled sql type " + sqlType);
                }
        }
        return 1;
    }

	public SqlParameter ToSqlParameter()
	{
		String name="@p"+pos;
		SqlDbType sqlDbType=SqlDbType.VarChar;
		switch (sqlType) 
		{
			case Types.VARCHAR:
			{
				sqlDbType=SqlDbType.VarChar;
				break;
			}
			case Types.INTEGER:
			{
				sqlDbType=SqlDbType.Int;
				break;
			}
			case Types.DOUBLE:
			{
				sqlDbType=SqlDbType.Float;
				break;
			}
			case Types.SMALLINT:
			{
				sqlDbType=SqlDbType.SmallInt;
				break;
			}
			case Types.TIME:
			{
				sqlDbType=SqlDbType.Timestamp;
				break;
			}
			case Types.TIMESTAMP:
			{
				sqlDbType=SqlDbType.Timestamp;
				break;
			}
			case Types.BIGINT:
			{
				sqlDbType=SqlDbType.BigInt;
				break;
			}
			case Types.FLOAT:
			{
				sqlDbType=SqlDbType.Float;
				break;
			}
			case Types.DATE:
			{
				sqlDbType=SqlDbType.DateTime;
				break;
			}
			default:
			{
				throw new Exception("Unhandled sql type " + sqlType);
			}
		}
		SqlParameter p=new SqlParameter(name,sqlDbType);
		p.Value=(value == null)?DBNull.Value:value;
		return p;
	}

    public override String ToString() {
        return "Param{" +
                "pos=" + pos +
                (name!=null? (", name=" + Utils.Dump(name)):"") +
                ", value=" + Utils.Dump(value) +
                ", sqlType=" + Types.ToSqlDbType(sqlType).ToString() + "("+sqlType+")"+
                "}";
    }
}
}

File: java/commons/src/main/java/org/vpc/neormf/commons/sql/SqlParam.java
package org.vpc.neormf.commons.sql;

import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;
import org.vpc.neormf.commons.util.Utils;

import java.io.Serializable;
import java.sql.*;
import org.vpc.neormf.commons.beans.BlobData;

/**
 * class presentation
 *
 * @author taha BEN SALAH (tbsalah)
 * @version 1.0
 * @copyrights (c) 2004, Vpc Open Source Foundary
 * @project Neormf
 * @creation on Date: 28 juil. 2004 Time: 11:20:37
 * @modification on ---- by -----
 * @modification on ---- by -----
 * @modification on ---- by -----
 */
public class SqlParam implements Serializable, Cloneable, SqlStatementParamsProvider {

    public static final int SQL_TYPE_EXTRA_BLOB = -7770;
    static final long serialVersionUID = 7306885846673824613L;
    private int pos;
    private String name;
    private Object value;
    private int sqlType;

    public SqlParam() {
    }

    public int getPos() {
        return pos;
    }

    public void setPos(int pos) {
        this.pos = pos;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Object getValue() {
        return value;
    }

    public void setValue(Object value) {
        this.value = value;
    }

    public int getSqlType() {
        return sqlType;
    }

    public void setSqlType(int sqlType) {
        this.sqlType = sqlType;
    }

    public int populateStatement(PreparedStatement ps, int startPosition) throws SQLException {
        int rpos = pos + startPosition - 1;
        if (value == null) {
            ps.setNull(rpos, sqlType);
            return 1;
        }
        switch (sqlType) {
            case Types.VARCHAR: {
                ps.setString(rpos, (String) value);
                break;
            }
            case Types.INTEGER: {
                ps.setInt(rpos, ((Integer) value).intValue());
                break;
            }
            case Types.DOUBLE: {
                ps.setDouble(rpos, ((Double) value).doubleValue());
                break;
            }
            case Types.SMALLINT: {
                ps.setShort(rpos, ((Short) value).shortValue());
                break;
            }
            case Types.TIME: {
                ps.setTime(rpos, ((Time) value));
                break;
            }
            case Types.TIMESTAMP: {
                ps.setTimestamp(rpos, ((Timestamp) value));
                break;
            }
            case Types.BIGINT: {
                ps.setLong(rpos, ((Long) value).longValue());
                break;
            }
            case Types.FLOAT: {
                ps.setFloat(rpos, ((Float) value).floatValue());
                break;
            }
            case Types.DATE: {
                ps.setDate(rpos, ((Date) value));
                break;
            }
            case SQL_TYPE_EXTRA_BLOB: {
                try {
                    SQLUtils.writeBlobData((BlobData) value, ps, pos);
                    break;
                } catch (IOException ex) {
                    Logger.getLogger(SqlParam.class.getName()).log(Level.SEVERE, null, ex);
                    throw new SQLException(ex.toString());
                }
            }

            default: {
                throw new IllegalArgumentException("Unhandled sql type " + sqlType);
            }
        }
        return 1;
    }

    public Object clone() {
        try {
            return super.clone();
        } catch (CloneNotSupportedException e) {
            throw new RuntimeException(e.getMessage());
        }
    }

    public String toString() {
        return "Param{" +
                "pos=" + pos +
                (name != null ? (", name=" + Utils.dump(name)) : "") +
                ", value=" + Utils.dump(value) +
                ", sqlType=" //+ SQLUtils.getTypeName(sqlType) 
                + "(" + sqlType + ")" +
                "}";
    }
}
